src/fr/hds/src/eu/kanade/tachiyomi/animeextension/fr/hds/Hds.kt

```diff
@@ -0,0 +1,75 @@
+package eu.kanade.tachiyomi.animeextension.fr.hds
+
+import eu.kanade.tachiyomi.animesource.model.SAnime
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.lib.filemoonextractor.FilemoonExtractor
+import eu.kanade.tachiyomi.lib.streamhidevidextractor.StreamHideVidExtractor
+import eu.kanade.tachiyomi.multisrc.dooplay.DooPlay
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.asJsoup
+import eu.kanade.tachiyomi.util.parallelCatchingFlatMapBlocking
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import uy.kohesive.injekt.injectLazy
+
+class Hds : DooPlay(
+    "fr",
+    "HDS",
+    "https://www.hds.quest",
+) {
+
+    private val json: Json by injectLazy()
+
+    // ============================== Popular ===============================
+    override fun popularAnimeRequest(page: Int) = GET("$baseUrl/tendance/page/$page/", headers)
+
+    override fun popularAnimeSelector() = latestUpdatesSelector()
+
+    override fun popularAnimeNextPageSelector() = "#nextpagination"
+
+    // =============================== Latest ===============================
+    override val supportsLatest = false
+
+    // =============================== Search ===============================
+    override fun searchAnimeNextPageSelector() = popularAnimeNextPageSelector()
+
+    // ============================== Filters ===============================
+    override fun genresListSelector() = ".genres.scrolling li a"
+
+    // =========================== Anime Details ============================
+    override fun animeDetailsParse(document: Document) = super.animeDetailsParse(document).apply {
+        if (document.select(".dt-breadcrumb li:nth-child(2)").text() == "Films") {
+            status = SAnime.COMPLETED
+        }
+    }
+
+    // ============================ Video Links =============================
+    @Serializable
+    data class VideoLinkDTO(@SerialName("embed_url") val url: String)
+
+    private val fileMoonExtractor by lazy { FilemoonExtractor(client) }
+    private val streamHideVidExtractor by lazy { StreamHideVidExtractor(client) }
+
+    override fun videoListParse(response: Response): List<Video> {
+        val document = response.use { it.asJsoup() }
```

Comment:
```suggestion
        val document = response.asJsoup()
```
No need to close it since using .string on a body will automatically close it